### PR TITLE
[KernelGen][Nvidia] Add special_softmax operator with Triton kernel

### PR DESCRIPTION
**Pull request by bwbwzzz**
## Summary
Adds a Triton kernel for `special_softmax`. Computes the softmax of `input` along dimension `dim`; alias of
`torch.nn.functional.softmax` exposed under `torch.special`.

## Testing
- Parametrized tests over multiple shapes and float16/float32/bfloat16 dtypes (15 cases)
- Validated against PyTorch reference on device side
- Tested on: Nvidia

## Repository / Branch
- Target repository: `flagos-ai/FlagGems`
- Base branch: `master`

## Performance
Test command: `pytest benchmark/test_special_softmax.py --level core` (NVIDIA H20)

### float16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 275.910 | 292.978 | 0.942 |
| 64, 64 | 0.007 | 0.006 | 1.115 |
| 4096, 4096 | 0.101 | 0.037 | 2.718 |
| 64, 512, 512 | 0.050 | 0.045 | 1.116 |
| 1024, 1024, 1024 | 2.903 | 1.718 | 1.690 |

### float32
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 391.837 | 305.654 | 1.282 |
| 64, 64 | 0.006 | 0.006 | 1.104 |
| 4096, 4096 | 0.100 | 0.047 | 2.116 |
| 64, 512, 512 | 0.052 | 0.051 | 1.010 |
| 1024, 1024, 1024 | 3.417 | 2.381 | 1.435 |

### bfloat16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 281.014 | 309.259 | 0.909 |
| 64, 64 | 0.007 | 0.006 | 1.025 |
| 4096, 4096 | 0.101 | 0.038 | 2.691 |
| 64, 512, 512 | 0.050 | 0.045 | 1.114 |
| 1024, 1024, 1024 | 2.871 | 1.785 | 1.608 |

**Arithmetic Mean Speedup: 1.458x**
## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (15 cases, --level core) | 1.458 | Primary |
| Tianshu | PASS | PASS (15 cases) | 1.64 | — |
| Muxi | PASS | PASS (15 cases) | 1.01 | — |
| Ascend | PASS | — | — | — |
| Hygon | PASS | PASS (15 cases) | 1.29 | — |

## Files Changed
- `src/flag_gems/ops/special_softmax.py`: Triton kernel implementation
- `tests/test_special_softmax.py`: Accuracy test
- `benchmark/test_special_softmax.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: Math, stage: alpha 5.4)

### CODE REVIEW

File: src/flag_gems/ops/special_softmax.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Generated by KernelGen: https://github.com/flagos-ai/KernelGen
```
Comment (bin913) [inline src/flag_gems/ops/special_softmax.py:15]: Please add copyright and license

File: src/flag_gems/ops/special_softmax.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+def special_softmax(self, dim, dtype=None):
+    logger.debug("GEMS SPECIAL_SOFTMAX")
+
+    if dtype is not None:
+        # Cast input to the specified dtype for computation
+        self = self.to(dtype)
```
Comment (bin913) [inline src/flag_gems/ops/special_softmax.py:29]: FlagGems also implements the `to` operator — would it be feasible to call FlagGems' `to` operator here instead?

Comment (bwbwzzz) [reply src/flag_gems/ops/special_softmax.py:29]: the torch.to operation is allowed to be used. If this to can fulfill the normal functionality, perhaps we don't need to use flaggems' to and introduce potential issues?  i dont know whether i should use gemsTo . thanks

File: src/flag_gems/ops/special_softmax.py (hunk 1 above)
Comment (bwbwzzz) [reply src/flag_gems/ops/special_softmax.py:15]: have added

**Review by bin913 — APPROVED** [review]
